src/dex/aave-v3-stata/aave-v3-stata.ts

```diff
@@ -0,0 +1,492 @@
+import { extractReturnAmountPosition } from '../../executor/utils';
+// import { IStaticATokenLM_ABI } from '@bgd-labs/aave-address-book';
+// slimmed down version of @bgd-labs/aave-address-book
+// required as version of web3-utils used is buggy
+//import IStaticATokenFactory_ABI from '../../abi/aave-v3-stata/StaticATokenFactory.json';
+
+export const TOKEN_LIST_CACHE_KEY = 'stata-token-list';
+const TOKEN_LIST_TTL_SECONDS = 24 * 60 * 60; // 1 day
+const TOKEN_LIST_LOCAL_TTL_SECONDS = 3 * 60 * 60; // 3h
+// const RAY = BigInt(1e27);
+const RAY = BigInt(`1${'0'.repeat(27)}`);
+
+export class AaveV3Stata
+  extends SimpleExchange
+  implements IDex<AaveV3StataData>
+{
+  readonly hasConstantPriceLargeAmounts = false;
+
+  readonly isFeeOnTransferSupported = false;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(AaveV3StataConfig);
+
+  logger: Logger;
+
+  // static readonly stata = new Interface(IStaticATokenLM_ABI);
+  static readonly stata = new Interface(TokenABI);
+
+  private state: Record<string, { blockNumber: number; rate: bigint }> = {};
+
+  constructor(
+    readonly network: Network,
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+    protected config = AaveV3StataConfig[dexKey][network],
+    protected adapters = Adapters[network],
+  ) {
+    super(dexHelper, dexKey);
+    this.logger = dexHelper.getLogger(dexKey);
+  }
+
+  // Initialize pricing is called once in the start of
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to
+  // implement this function
+  async initializePricing(blockNumber: number) {
+    let cachedTokenList = await this.dexHelper.cache.getAndCacheLocally(
+      this.dexKey,
+      this.network,
+      TOKEN_LIST_CACHE_KEY,
+      TOKEN_LIST_LOCAL_TTL_SECONDS,
+    );
+    if (cachedTokenList !== null) {
+      setTokensOnNetwork(this.network, JSON.parse(cachedTokenList));
+      return;
+    }
+
+    let tokenList = await fetchTokenList(
+      this.dexHelper.web3Provider,
+      blockNumber,
+      this.config.factoryAddress,
+      this.dexHelper.multiWrapper,
+    );
+
+    await this.dexHelper.cache.setex(
+      this.dexKey,
+      this.network,
+      TOKEN_LIST_CACHE_KEY,
+      TOKEN_LIST_TTL_SECONDS,
+      JSON.stringify(tokenList),
+    );
+
+    setTokensOnNetwork(this.network, tokenList);
+
+    // init state for all tokens as empty
+    tokenList.forEach(token => {
+      this.state[token.address] = {
+        blockNumber: 0,
+        rate: 0n,
+      };
+    });
+  }
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] ? this.adapters[side] : null;
+  }
+
+  private _getPoolIdentifier(srcToken: Token, destToken: Token): string {
+    return (
+      this.dexKey +
```

Comment:
please add `-` after `this.dexKey` to ease debugging in logs for this dex

Reply:
done